### ISSUE TITLE
Add `sync_until` method to `pcli::sync`

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/penumbra#376]
Currently, `pcli` has a method
```
pub async fn sync(opt: &Opt, state: &mut ClientStateFile) -> Result<()> { /* .. */ }
```
It would also be useful to have a method
```
pub async fn sync_until(opt: &Opt, state: &mut ClientStateFile, nc: note::Commitment) -> Result<()> { /* .. */ }
```
that would attempt to synchronize until the provided note commitment `nc` was observed in the wallet state.  This would let us wait on change notes.

In a later iteration, we could use Tendermint's websockets or something to be notified when new blocks arrive, but for now, we could do something much simpler, along the lines of:
```
pub async fn sync_until(opt: &Opt, state: &mut ClientStateFile, nc: note::Commitment) -> Result<()> {
    loop {
        sync(opt, state).await?;
        if /* nc in state */ { return Ok(()) }
        tokio::time::sleep(Duration::from_secs(3)).await;
    }
}
```
This doesn't need to have a timeout logic built in, since the call to `sync_until` can be wrapped in a `tokio::time::timeout`, which is much simpler than using an internal select.

**Comment by plaidfinch** [comment]
Are there situations where we'd like to use this to await more than one note at the same time? Should it take `&[note::Commitment]`?

**Comment by hdevalence** [comment]
I don't think so, if the method returns immediately when the note is already present, you can get the same effect by looping and waiting for each note individually without extra wait time (some layer notes might be seen while waiting for earlier ones).

**Comment by hdevalence** [comment]
Closing this because it may become stale by the time we revisit the wallet code.

**Comment by hdevalence** [comment]
Closed in favor of #935
